app/Models/Extensions/ConfigsHas.php

```diff
@@ -27,31 +29,26 @@ public static function hasExiftool(): bool
 		// 1: Exiftool is available
 		// 2: Not yet tested if exiftool is available
 
-		$has_exiftool = intval(self::get_value('has_exiftool'));
+		$has_exiftool = intval(self::get_value('has_exiftool', 2));
 
 		// value not yet set -> let's see if exiftool is available
-		if ($has_exiftool == 2) {
+		if ($has_exiftool === 2) {
 			try {
 				$path = exec('command -v exiftool');
-				if ($path == '') {
-					self::set('has_exiftool', 0);
-					$has_exiftool = false;
-				} else {
-					self::set('has_exiftool', 1);
-					$has_exiftool = true;
-				}
-			} catch (Exception $e) {
-				self::set('has_exiftool', 0);
-				$has_exiftool = false;
-				Logs::warning(__METHOD__, __LINE__, 'exec is disabled, has_exiftool has been set to 0.');
+			} catch (\Exception $e) {
+				$path = '';
+				Handler::reportSafely(new ExternalComponentMissingException('could not find exiftool, `has_exiftool` will be set to 0', $e));
```

Comment:
Nitpicking mode **on** :wink:

```suggestion
				Handler::reportSafely(new ExternalComponentMissingException('could not find exiftool; `has_exiftool` will be set to 0', $e));
```